cpp/src/codegen/like_holder.cc

```diff
@@ -24,6 +22,39 @@ namespace gandiva {
 namespace helpers {
 #endif
 
+// Short-circuit pattern matches for the two common sub cases :
+// - starts_with and ends_with.
+const FunctionNode LikeHolder::TryOptimize(const FunctionNode &node) {
+  static std::string starts_with_pattern(R"((\w|\s)*\.\*)");
```

Comment:
[just for my understanding]
why are we not optimizing for ?(one or zero) +(one or more) too..

also what is the difference in performance because of this change..

Reply:
SQL like supports only % (0 or more) and _ (exactly one).

https://www.w3schools.com/sql/sql_like.asp

I'll check the perf of '_' and short-circuit that also, if required.

For 1M matches, re2 takes 113ms and the short-circuit takes 12ms. In comparison, java takes 68ms.